projects/pastanaga-angular/src/lib/side-nav/side-nav.component.ts

```diff
@@ -37,46 +38,52 @@ export class SideNavComponent implements OnInit, AfterContentInit {
         return !!this._visible;
     }
     set visible(value: boolean) {
-        this._visible = coerceBooleanProperty(value);
+        if (this._mode !== 'desktop' && !this.modeChanged) {
+            this.triggerAnimation(value);
+        } else {
+            this.modeChanged = false;
+            this._visible = coerceBooleanProperty(value);
+        }
+    }
+    @Input()
+    get mode(): string {
+        return this._mode;
     }
-    @Input() toggle?: Subject<boolean>;
+    set mode(value: string) {
+        this.modeChanged = value !== this._mode;
+        this._mode = value;
+    }
+    @Output() close: Subject<void> = new Subject<void>();
```

Comment:
should be an EventEmitter, not a subject

Reply:
isn't the same? i will change it but i'm not totally sure why i should use one instead of the other one